### PR TITLE
Documentation for customizing snap-in configuration page 

### PR DESCRIPTION
**Pull request by codeon**
ISS-98485
ISS-100260
ISS-98417

### CODE REVIEW

File: fern/docs/pages/references/snap-in-configuration.mdx [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+          ],
+          "type": "card"
+        }
+      ]
+    }
+  }
+}
+```
+
+In this example, the snap-kit renders a dropdown select for choosing between "Ticket" and "Conversation," along with a "Next" button. When the "Next" button is clicked, the `user_snap_in_configuration_handler` function is invoked to process the user's selection.
```
Comment (bc-devrev) [inline fern/docs/pages/references/snap-in-configuration.mdx:110]: Use code phrases to refer to items in the code sample. Use bold for UI elements like buttons.
```suggestion
In this example, the snap-kit renders a dropdown select for choosing between `Ticket` and `Conversation`, along with a `Next` button. When the **Next** button is clicked, the `user_snap_in_configuration_handler` function is invoked to process the user's selection.
```

File: fern/docs/pages/references/snap-in-configuration.mdx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+In the snap-kit action handler, the `event.payload.action.id` can be used to determine the form being submitted and call the `snap-ins.update` API to update the configuration.
+
+## Update snap-in inputs (BETA)
+
+Updates the inputs of a snap-in based on the inputs defined in the snap-in configuration.
+
+**Note: This endpoint is currently in BETA and may be subject to change in the future. Reach out to us via PLuG to subscribe to changes to BETA endpoints.**
```
Comment (bc-devrev) [inline fern/docs/pages/references/snap-in-configuration.mdx:117]:
```suggestion
**Note: This endpoint is currently in beta and may be subject to change in the future. Reach out to us via PLuG to subscribe to changes to beta endpoints.**
```

File: fern/docs/pages/references/snap-in-configuration.mdx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+      ]
+    }
+  }
+}
+```
+
+In this example, the snap-kit renders a dropdown select for choosing between "Ticket" and "Conversation," along with a "Next" button. When the "Next" button is clicked, the `user_snap_in_configuration_handler` function is invoked to process the user's selection.
+In the snap-kit action handler, the `event.payload.action.id` can be used to determine the form being submitted and call the `snap-ins.update` API to update the configuration.
+
+## Update snap-in inputs (BETA)
```
Comment (bc-devrev) [inline fern/docs/pages/references/snap-in-configuration.mdx:113]:
```suggestion
## Update snap-in inputs (beta)
```

File: fern/docs/pages/references/snap-in-configuration.mdx [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+The DevRev snap-ins platform allows developers to define custom configuration pages for their snap-ins. 
+
+While the default configuration page automatically renders input fields for keyrings and inputs, there may be cases where a custom configuration page is more suitable:
+
+- Improved user experience: Developers can design the configuration page to provide a more intuitive and streamlined experience for users setting up the snap-in.
+- Advanced input handling: Custom configuration pages enable developers to handle complex input scenarios, such as fetching data from external systems to populate dropdown options or validating user input.
+- Branding and styling: Developers can align the configuration page with their Snap-in's branding and style guidelines, ensuring a consistent look and feel.
+
+## Defining Custom Configuration Pages
```
Comment (bc-devrev) [inline fern/docs/pages/references/snap-in-configuration.mdx:11]: Sentence caps on titles. https://www.notion.so/devrev/Which-words-in-a-heading-or-title-should-be-capitalized-81984d54dd414af5b343bce1327c05bd?pvs=4
```suggestion
## Defining custom configuration pages
```

File: fern/docs/pages/references/snap-in-configuration.mdx [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+In this example, the snap-kit renders a dropdown select for choosing between "Ticket" and "Conversation," along with a "Next" button. When the "Next" button is clicked, the `user_snap_in_configuration_handler` function is invoked to process the user's selection.
+In the snap-kit action handler, the `event.payload.action.id` can be used to determine the form being submitted and call the `snap-ins.update` API to update the configuration.
+
+## Update snap-in inputs (BETA)
+
+Updates the inputs of a snap-in based on the inputs defined in the snap-in configuration.
+
+**Note: This endpoint is currently in BETA and may be subject to change in the future. Reach out to us via PLuG to subscribe to changes to BETA endpoints.**
+
+### Request Payload
```
Comment (bc-devrev) [inline fern/docs/pages/references/snap-in-configuration.mdx:119]:
```suggestion
### Request payload
```

File: fern/docs/pages/references/snap-in-configuration.mdx [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+In this example, the `updateSnapInInputs` function takes the `snapInId` and `inputsValues` as parameters. The `inputsValues` object should contain the input names and their corresponding values as defined in the snap-in configuration.
+
+The function constructs the payload object with the `snapInId` and `inputsValues`, and then calls the `updateSnapInInputs` function to make the POST request to the `snap-ins.update` endpoint.
+
+If the request is successful, it returns a success response with a status code of 200 and a message indicating that the snap-in was updated successfully.
+
+If an error occurs, it catches the error and returns an error response with an appropriate status code and an error message containing the error details.
+
+Please keep in mind that this endpoint is currently in BETA, and its functionality or parameters may change in future updates.
```
Comment (bc-devrev) [inline fern/docs/pages/references/snap-in-configuration.mdx:207]:
```suggestion
<Note>
Note: This endpoint is currently in beta, and its functionality or parameters may change in future updates.
</Note>
```

**Review by bc-devrev — APPROVED** [review]
Some capitalization and styling suggestions but fundamentally ok.

**Review by umang-malik — APPROVED** [review]

**Pull request merged**